src/pthread-main.js

```diff
@@ -0,0 +1,96 @@
+// Pthread Web Worker startup routine:
+// This is the entry point file that is loaded first by each Web Worker
+// that executes pthreads on the Emscripten application.
+
+// All pthreads share the same Emscripten HEAP as SharedArrayBuffer
+// with the main execution thread.
+var buffer;
+
+var threadBlock = 0; // Info area for this thread in Emscripten HEAP (shared). If zero, this worker is not currently hosting an executing pthread.
```

Comment:
is this related or similar to to thread local storage?

The term "block" seems possibly confusing here, as this isn't something blocking on something else, but a block of info. How about "header" or "info" or "headerPtr" etc.?